### ISSUE TITLE
[DateRangePicker] customizable width / height

### ISSUE THREAD

**Issue by LeoLozes** [mui/mui-x#4565]
We are very interested to invest in material-UI X and using the new Grid & Calendar components.
I was playing a bit with the calendar yesterday and I find it great, specially the static mode of the ranged calendar (with the awesome selection), it's something that would be really useful to us. 

Unfortunately we are trying to build something like this, with a width that adapts to the parent component, and a calendar that is in a whole a lot bigger than the current one:
![image](https://user-images.githubusercontent.com/11133817/100994229-aa90ad80-3556-11eb-9639-54581f4bdca5.png)

I tried to fiddle with the styling but didn't find any way to get it to be wider without breaking everything.
Is this something you would consider in the future? Or the plan is to stick with the "fixed" material design?

Thank you again for the great library you're building, it helps us a lot!

**Comment by hmmftg** [comment]
Hi, I've implemented a fullscreen calendar using mui datepicker and I think you can add it in example page or as demo in calendar systems page, it will help developers from tring to design outlook like calendar pages for sure, how can I send the sample code? should I add a PR?

**Comment by LukasTy** [comment]
@hmmftg Yes, please create a PR adding a new documentation demo on a relevant page (depending on what component is used). 🙏

**Comment by hmmftg** [comment]
@LukasTy can you help me with my [PR](https://github.com/mui/mui-x/pull/18849)? as circle ci is blocked in my country I couldn't  view the error in that stage
